chain/src/chain.rs

```diff
@@ -280,74 +289,86 @@ impl<CI: ChainIndex + 'static> ChainService<CI> {
         }
     }
 
-    fn get_forks(
+    pub(crate) fn get_forks(
         &self,
         current_tip_number: BlockNumber,
         new_tip_block: &Block,
         new_tip_ext: BlockExt,
     ) -> Option<Fork> {
-        let mut number = new_tip_block.header().number() - 1;
+        let new_tip_number = new_tip_block.header().number();
         let mut old_blocks = Vec::new();
         let mut new_blocks = Vec::new();
         let mut open_exts = Vec::new();
 
         if new_tip_ext.valid.is_none() {
             open_exts.push(new_tip_ext);
         } else {
-            // must be Some(false)
+            // txs in block are invalid
             return None;
         }
 
-        let mut is_open = true;
+        let mut is_new_record = true;
 
-        // The old fork may longer than new fork
-        if number < current_tip_number {
-            for n in number..=current_tip_number {
-                let hash = self.shared.block_hash(n).unwrap();
+        // new_blocks = forks[latest_common + 1 .. new_tip]
+        // old_blocks = chain[latest_common + 1 .. old_tip]
+        new_blocks.push(new_tip_block.clone());
+        let mut index = new_tip_number - 1;
+        let mut index_hash = new_tip_block.header().parent_hash().clone();
```

Comment:
Use some name like `new_chain_index` to distinguish it with the `index` variable, those two value is confused.